model/task/task.go

```diff
@@ -356,16 +356,38 @@ func (t *Task) SetOverrideDependencies(userID string) error {
 }
 
 func (t *Task) AddDependency(d Dependency) error {
+	query := bson.M{IdKey: t.Id}
+	update := bson.M{
+		"$push": bson.M{
+			DependsOnKey: d,
+		},
+	}
+	for _, existingDependency := range t.DependsOn {
+		if existingDependency.TaskId == d.TaskId && existingDependency.Status == d.Status {
+			// If the dependency in the DB is attainable and we want to set it unattainable, update dependency.
+			// Otherwise, no update needed.
+			if !existingDependency.Unattainable && d.Unattainable {
+				query[DependsOnKey] = bson.M{
+					"$elemMatch": bson.M{
+						DependencyTaskIdKey: d.TaskId,
+						DependencyStatusKey: d.Status,
+					},
+				}
+				query[bsonutil.GetDottedKeyName(DependsOnKey, DependencyStatusKey)] = d.Status
```

Comment:
I don't understand what this adds.
From [here](https://docs.mongodb.com/manual/reference/operator/update/positional/#update-embedded-documents-using-multiple-field-matches) it looks like the first predicate is sufficient.

Reply:
Ah I forgot to remove this from previous query attempts, but it's now moot :)